core/lib/dal/src/tokens_web3_dal.rs

```diff
@@ -31,17 +56,45 @@ impl TokensWeb3Dal<'_, '_> {
         .fetch_all(self.storage.conn())
         .await?;
 
-        Ok(records
-            .into_iter()
-            .map(|record| TokenInfo {
-                l1_address: Address::from_slice(&record.l1_address),
-                l2_address: Address::from_slice(&record.l2_address),
-                metadata: TokenMetadata {
-                    name: record.name,
-                    symbol: record.symbol,
-                    decimals: record.decimals as u8,
-                },
-            })
-            .collect())
+        Ok(records.into_iter().map(Into::into).collect())
+    }
+
+    /// Returns information about all tokens.
+    pub async fn get_all_tokens(
```

Comment:
- Another way to implement this method would be to add `tokens.miniblock_number` column. AFAIU, it'd require a programmatic migration to fill data because there's no way to compute a hashed key from address + key inside an SQL query.
- Or we can not filter tokens on the server side at all, and use client-side filtering.